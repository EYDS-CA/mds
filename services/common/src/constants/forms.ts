export enum FORM {
  // projects
  ADD_EDIT_PROJECT_SUMMARY = "ADD_EDIT_PROJECT_SUMMARY",
  ADD_MINE_MAJOR_APPLICATION = "ADD_MINE_MAJOR_APPLICATION",
  // reports
  VIEW_EDIT_REPORT = "VIEW_EDIT_REPORT",
  // permits
  REVOKE_DIGITAL_CREDENTIAL = "REVOKE_DIGITAL_CREDENTIAL",
  REQUEST_REPORT = "REQUEST_REPORT",
  ADD_REPORT_TO_PERMIT_CONDITION = "ADD_REPORT_TO_PERMIT_CONDITION",
  ADD_PERMIT_CONDITION_CATEGORY = "ADD_PERMIT_CONDITION_CATEGORY",
  EDIT_PERMIT_CONDITION = "EDIT_PERMIT_CONDITION",
  // tailings
  ADD_TAILINGS = "ADD_TAILINGS",
  // documents
  REPLACE_DOCUMENT = "REPLACE_DOCUMENT",
  DELETE_DOCUMENT = "DELETE_DOCUMENT",
  ARCHIVE_DOCUMENT = "ARCHIVE_DOCUMENT",
  EDIT_HELP_GUIDE = "EDIT_HELP_GUIDE",
  INLINE_EDIT_PERMIT_CONDITION_CATEGORY = "INLINE_EDIT_PERMIT_CONDITION_CATEGORY",
  PERMIT_CONDITION_REVIEW_ASSIGNMENT = "PERMIT_CONDITION_REVIEW_ASSIGNMENT",
  UPDATE_MAJOR_MINE_APPLICATION = "UPDATE_MAJOR_MINE_APPLICATION",
  UPDATE_IRT = "UPDATE_IRT"
}
