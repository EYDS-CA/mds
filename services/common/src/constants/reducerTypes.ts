// Misc
export const AUTHENTICATION = "AUTHENTICATION";
export const STATIC_CONTENT = "STATIC_CONTENT";
export const MODAL = "MODAL";
export const DOCUMENT_VIEWER = "DOCUMENT_VIEWER";
export const SYSTEM_FLAG = "SYSTEM_FLAG";

// Mines
export const MINES = "MINES";

// Parties
export const PARTIES = "PARTIES";

// OrgBook
export const ORGBOOK = "ORGBOOK";

// Compliance Orders
export const COMPLIANCE = "COMPLIANCE";

// MineSpace
export const MINESPACE = "MINESPACE";

// Permits
export const PERMITS = "PERMITS";

// Explosive Storage & Use Permits
export const EXPLOSIVES_PERMITS = "EXPLOSIVES_PERMITS";

// Search
export const SEARCH = "SEARCH";

// Variances
export const VARIANCES = "VARIANCES";

// Projects
export const PROJECTS = "PROJECTS";
export const PROJECT_SUMMARIES = "PROJECT_SUMMARIES";
export const INFORMATION_REQUIREMENTS_TABLE = "INFORMATION_REQUIREMENTS_TABLE";

// Core Users
export const USERS = "USERS";

// Incidents
export const INCIDENTS = "INCIDENTS";

// Work Information
export const MINE_WORK_INFORMATIONS = "MINE_WORK_INFORMATIONS";

// Reports
export const REPORTS = "REPORTS";

// Notices of Work
export const NOTICE_OF_WORK = "NOTICE_OF_WORK";

// Securities
export const SECURITIES = "SECURITIES";

// EPIC Information
export const EXTERNAL_AUTHS = "EXTERNAL_AUTHS";

// Notices of Departure
export const NOTICES_OF_DEPARTURE = "NOTICES_OF_DEPARTURE";

// Activities
export const ACTIVITIES = "ACTIVITIES";

// Tailings Storage Facilities
export const TAILINGS = "TAILINGS";

// Dams
export const DAMS = "DAMS";


//Verifiable Credentials
export const VERIFIABLE_CREDENTIALS = "VERIFIABLE_CREDENTIALS";
