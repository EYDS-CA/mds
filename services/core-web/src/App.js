import React from "react";
import { BrowserRouter } from "react-router-dom";
import { hot } from "react-hot-loader";
import Routes from "./routes/Routes";
import ModalWrapper from "@/components/common/wrappers/ModalWrapper";
import DocumentViewer from "@/components/syncfusion/DocumentViewer";
import ScrollToTopWrapper from "@/components/common/wrappers/ScrollToTopWrapper";
import configureStore from "./store/configureStore";

export const store = configureStore();

const App = () => (
  <BrowserRouter basename={process.env.BASE_PATH}>
    <ScrollToTopWrapper>
      <>
        <Routes />
        <ModalWrapper />
        <DocumentViewer />
      </>
    </ScrollToTopWrapper>
  </BrowserRouter>
);

export default hot(module)(App);
