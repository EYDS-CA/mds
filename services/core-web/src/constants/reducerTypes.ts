// Documents
export const DOCUMENTS = "DOCUMENTS";

// Alerts
export const MINE_ALERTS = "MINE_ALERTS";