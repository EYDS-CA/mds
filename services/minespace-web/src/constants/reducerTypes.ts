// User Dashboard Info
export const USER_MINE_INFO = "USER_MINE_INFO";
