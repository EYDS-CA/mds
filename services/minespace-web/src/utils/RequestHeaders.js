import keycloak from "@/keycloak";
/**
 * Utility class for various request Headers.
 */

export const createRequestHeader = () => ({
  headers: {
    "Access-Control-Allow-Origin": "*",
    Authorization: `Bearer ${keycloak.token}`,
  },
});
