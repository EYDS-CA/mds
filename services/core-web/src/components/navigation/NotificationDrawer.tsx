import React, { RefObject, useEffect, useRef, useState } from "react";
import { Badge, Button, Col, Row, Tabs, Typography } from "antd";
import { BellOutlined } from "@ant-design/icons";
import { useSelector, useDispatch } from "react-redux";
import {
  fetchActivities,
  markActivitiesAsRead,
} from "@mds/common/redux/actionCreators/activityActionCreator";
import { formatDateTimeUserTz } from "@common/utils/helpers";
import { getActivities } from "@mds/common/redux/selectors/activitySelectors";
import { getUserInfo } from "@mds/common/redux/selectors/authenticationSelectors";
import { useHistory } from "react-router-dom";
import { storeActivities } from "@mds/common/redux/actions/activityActions";
import {
  MINE_TAILINGS_DETAILS,
  NOTICE_OF_DEPARTURE,
  EDIT_PROJECT_SUMMARY,
  EDIT_PROJECT,
  VIEW_MINE_INCIDENT,
  REPORT_VIEW_EDIT,
} from "@/constants/routes";
import { IActivity } from "@mds/common/interfaces";

const NotificationDrawer = () => {
  const [open, setOpen] = useState(false);
  const history = useHistory();
  const dispatch = useDispatch();
  const userInfo = useSelector(getUserInfo);
  const activities = useSelector(getActivities)

  const handleMarkAsRead = async (guid = null) => {
    if (guid) {
      await dispatch(markActivitiesAsRead([guid]));
    }

    const readActivities = activities.map((activity) => {
      if (guid) {
        if (activity.notification_guid === guid) {
          return {
            ...activity,
            notification_read: true,
          };
        }
        return activity;
      }
      return {
        ...activity,
        notification_read: true,
      };
    });
    await dispatch(storeActivities({
      records: readActivities,
      totalActivities: readActivities.length,
    }));
  };

  const outsideClickHandler = (ref: RefObject<HTMLDivElement>) => {
    useEffect(() => {
      const handleClickOutside = (event) => {
        if (open) {
          if (ref.current && !ref.current.contains(event.target)) {
            setOpen(!open);
          }
        }
      };
      document.addEventListener("mousedown", handleClickOutside);
      return () => {
        document.removeEventListener("mousedown", handleClickOutside);
      };
    }, [ref, open]);
  };

  const handleCollapse = () => {
    setOpen(!open);
  };

  const handleMarkAllAsRead = async () => {
    const activitiesToMarkAsRead = activities.reduce((acc, activity) => {
      if (activity.notification_read === false) {
        acc.push(activity.notification_guid);
      }
      return acc;
    }, []);

    if (activitiesToMarkAsRead.length > 0) {
      dispatch(markActivitiesAsRead(activitiesToMarkAsRead));
    }
    await handleMarkAsRead();
    handleCollapse();
  };

  const navigationHandler = async (notification: IActivity) => {
    switch (notification.notification_document.metadata.entity) {
      case "MineReport":
        return REPORT_VIEW_EDIT.dynamicRoute(
          notification.notification_document.metadata.mine.mine_guid,
          notification.notification_document.metadata.entity_guid
        );
      case "NoticeOfDeparture":
        return NOTICE_OF_DEPARTURE.dynamicRoute(
          notification.notification_document.metadata.mine.mine_guid,
          notification.notification_document.metadata.entity_guid
        );
      case "MineIncident":
        return VIEW_MINE_INCIDENT.dynamicRoute(
          notification.notification_document.metadata.mine.mine_guid,
          notification.notification_document.metadata.entity_guid
        );
      case "ProjectSummary":
        return EDIT_PROJECT_SUMMARY.dynamicRoute(
          notification.notification_document.metadata.project.project_guid,
          notification.notification_document.metadata.entity_guid
        );
      case "InformationRequirementsTable":
        return EDIT_PROJECT.dynamicRoute(
          notification.notification_document.metadata.project.project_guid,
          "information-requirements-table"
        );
      case "MajorMineApplication":
        return EDIT_PROJECT.dynamicRoute(
          notification.notification_document.metadata.project.project_guid,
          "app"
        );
      case "EngineerOfRecord":
        return MINE_TAILINGS_DETAILS.dynamicRoute(
          notification.notification_document.metadata.mine_tailings_storage_facility
            .mine_tailings_storage_facility_guid,
          notification.notification_document.metadata.mine.mine_guid,
          "engineer-of-record"
        );
      case "QualifiedPerson":
        return MINE_TAILINGS_DETAILS.dynamicRoute(
          notification.notification_document.metadata.mine_tailings_storage_facility
            .mine_tailings_storage_facility_guid,
          notification.notification_document.metadata.mine.mine_guid,
          "qualified-person"
        );
      case "DocumentManagement":
        return EDIT_PROJECT.dynamicRoute(
          notification.notification_document.metadata.entity_guid,
          "documents"
        );
      default:
        return null;
    }
  };

  const activityClickHandler = async (notification: IActivity) => {
    await handleMarkAsRead(notification.notification_guid);
    handleCollapse();
    const route = await navigationHandler(notification);
    history.push(route);
  };

  const modalRef = useRef(null);
  outsideClickHandler(modalRef);

  useEffect(() => {
    const fetchData = async () => {
      await dispatch(fetchActivities(userInfo?.preferred_username));
    };

    if (userInfo?.preferred_username) {
      fetchData();
    }
  }, [userInfo?.preferred_username]);

  return (
    <div ref={modalRef}>
      <Button
        onClick={handleCollapse}
        type="text"
        className={`notification-button ${open ? "notification-button-open" : ""}`}
        icon={
          <Badge
            className="notification-badge"
            count={activities?.filter((act) => !act?.notification_read).length || 0}
          >
            <BellOutlined className="notification-icon" />
          </Badge>
        }
      />
      <div className={`notification-drawer ${open ? "notification-drawer-open" : ""}`}>
        <Tabs
          defaultActiveKey="1"
          tabBarStyle={{ backgroundColor: "#FFF" }}
          className="notification-tabs"
        >
          <Tabs.TabPane
            className="notification-tab-pane"
            tab={<Typography className="notification-tab-header">Mine Activity</Typography>}
            key="1"
          >
            <div className="notification-button-all-container">
              <Button
                className="notification-button-all"
                size="small"
                type="text"
                onClick={() => handleMarkAllAsRead()}
              >
                Mark all as read
              </Button>
            </div>
            {(activities || [])?.map((activity) => (
              <div className="notification-list-item" key={activity.notification_guid}>
                <div className={!activity.notification_read ? "notification-dot" : ""} />
                <div tabIndex={0} role="button" onClick={() => activityClickHandler(activity)}>
                  <Typography.Text>{activity.notification_document?.message}</Typography.Text>
                  <Row className="items-center margin-small" gutter={6}>
                    <Col>
                      <Typography.Text className="notification-info-text">
                        {activity.notification_document?.metadata?.mine?.mine_name}
                      </Typography.Text>
                    </Col>
                    {activity.notification_document?.metadata?.permit && (
                      <>
                        <Col>
                          <div className="notification-separator" />
                        </Col>
                        <Col>
                          <Typography.Text className="notification-info-text">
                            {activity.notification_document?.metadata?.permit?.permit_no}
                          </Typography.Text>
                        </Col>
                        <Col>
                          <div className="notification-separator" />
                        </Col>
                      </>
                    )}
                    <Col>
                      <Typography.Text className="notification-info-text">
                        {formatDateTimeUserTz(activity.create_timestamp)}
                      </Typography.Text>
                    </Col>
                  </Row>
                </div>
              </div>
            ))}
          </Tabs.TabPane>
        </Tabs>
      </div>
    </div>
  );
};

export default NotificationDrawer;
