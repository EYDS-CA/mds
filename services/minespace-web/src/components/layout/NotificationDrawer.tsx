import React, { RefObject, useEffect, useRef, useState } from "react";
import { Badge, Button, Col, Row, Tabs, Typography } from "antd";
import BellOutlined from "@ant-design/icons/BellOutlined";
import { useDispatch, useSelector } from "react-redux";
import {
  fetchActivities,
  markActivitiesAsRead,
} from "@mds/common/redux/actionCreators/activityActionCreator";
import { formatDateTime } from "@common/utils/helpers";
import { getActivities } from "@mds/common/redux/selectors/activitySelectors";
import { getUserInfo } from "@mds/common/redux/selectors/authenticationSelectors";
import { storeActivities } from "@mds/common/redux/actions/activityActions";
import { useHistory } from "react-router-dom";
import {
  EDIT_MINE_INCIDENT,
  EDIT_PROJECT_SUMMARY,
  EDIT_TAILINGS_STORAGE_FACILITY,
  MINE_DASHBOARD,
  REVIEW_INFORMATION_REQUIREMENTS_TABLE,
  REVIEW_MAJOR_MINE_APPLICATION,
  REPORT_VIEW_EDIT,
} from "@/constants/routes";
import { IActivity } from "@mds/common/interfaces";

const NotificationDrawer = () => {
  const [open, setOpen] = useState(false);
  const history = useHistory();
  const dispatch = useDispatch();
  const userInfo = useSelector(getUserInfo);
  const activities = useSelector(getActivities);

  const handleMarkAsRead = async (guid: string = null) => {
    if (guid) {
      await dispatch(markActivitiesAsRead([guid]));
    }

    const readActivities = activities.map((activity) => {
      if (guid) {
        if (activity.notification_guid === guid) {
          return {
            ...activity,
            notification_read: true,
          };
        }
        return activity;
      }
      return {
        ...activity,
        notification_read: true,
      };
    });
    await dispatch(
      storeActivities({
        records: readActivities,
        totalActivities: readActivities.length,
      })
    );
  };

  const outsideClickHandler = (ref: RefObject<HTMLDivElement>) => {
    useEffect(() => {
      const handleClickOutside = (event) => {
        if (open) {
          if (ref.current && !ref.current.contains(event.target)) {
            setOpen(!open);
          }
        }
      };
      document.addEventListener("mousedown", handleClickOutside);
      return () => {
        document.removeEventListener("mousedown", handleClickOutside);
      };
    }, [ref, open]);
  };

  const handleCollapse = () => {
    setOpen(!open);
  };

  const handleMarkAllAsRead = async () => {
    const activitiesToMarkAsRead = activities.reduce((acc, activity) => {
      if (activity.notification_read === false) {
        acc.push(activity.notification_guid);
      }
      return acc;
    }, []);

    if (activitiesToMarkAsRead.length > 0) {
      dispatch(markActivitiesAsRead(activitiesToMarkAsRead));
    }
    await handleMarkAsRead();
    handleCollapse();
  };

  const navigationHandler = (notification: IActivity) => {
    switch (notification.notification_document.metadata.entity) {
      case "MineReport":
        const { metadata } = notification.notification_document;
        return {
          route: REPORT_VIEW_EDIT.dynamicRoute(metadata.mine.mine_guid, metadata.entity_guid),
          state: { isEditMode: true },
        };
      case "NoticeOfDeparture":
        return {
          route: MINE_DASHBOARD.dynamicRoute(
            notification.notification_document.metadata.mine.mine_guid,
            "nods"
          ),
          search: `?${new URLSearchParams({
            nod: notification.notification_document.metadata.entity_guid,
          })}`,
          state: {},
        };
      case "MineIncident":
        return {
          route: EDIT_MINE_INCIDENT.dynamicRoute(
            notification.notification_document.metadata.mine.mine_guid,
            notification.notification_document.metadata.entity_guid
          ),
          state: {},
        };
      case "ProjectSummary":
        return {
          route: EDIT_PROJECT_SUMMARY.dynamicRoute(
            notification.notification_document.metadata.project.project_guid,
            notification.notification_document.metadata.entity_guid
          ),
          state: {},
        };
      case "InformationRequirementsTable":
        return {
          route: REVIEW_INFORMATION_REQUIREMENTS_TABLE.dynamicRoute(
            notification.notification_document.metadata.project.project_guid,
            notification.notification_document.metadata.entity_guid
          ),
          state: { current: 2 },
        };
      case "MajorMineApplication":
        return {
          route: REVIEW_MAJOR_MINE_APPLICATION.dynamicRoute(
            notification.notification_document.metadata.project.project_guid,
            notification.notification_document.metadata.entity_guid
          ),
          state: { current: 2 },
        };
      case "EngineerOfRecord":
        return {
          route: EDIT_TAILINGS_STORAGE_FACILITY.dynamicRoute(
            notification.notification_document.metadata.mine_tailings_storage_facility
              .mine_tailings_storage_facility_guid,
            notification.notification_document.metadata.mine.mine_guid,
            "engineer-of-record"
          ),
          state: {},
        };
      case "QualifiedPerson":
        return {
          route: EDIT_TAILINGS_STORAGE_FACILITY.dynamicRoute(
            notification.notification_document.metadata.mine_tailings_storage_facility
              .mine_tailings_storage_facility_guid,
            notification.notification_document.metadata.mine.mine_guid,
            "qualified-person"
          ),
          state: {},
        };
      default:
        return null;
    }
  };

  const activityClickHandler = async (notification: IActivity) => {
    await handleMarkAsRead(notification.notification_guid);
    handleCollapse();
    const routeSpecifics = navigationHandler(notification);
    const route = {
      pathname: routeSpecifics?.route,
      search: routeSpecifics?.search,
      state: { ...(routeSpecifics?.state ?? {}) },
    };
    history.push(route);
  };

  const modalRef = useRef(null);
  outsideClickHandler(modalRef);

  useEffect(() => {
    const fetchData = async () => {
      await dispatch(fetchActivities(userInfo?.preferred_username));
    };

    if (userInfo?.preferred_username) {
      fetchData();
    }
  }, [userInfo?.preferred_username]);

  return (
    <div ref={modalRef}>
      <Button
        onClick={handleCollapse}
        type="text"
        className={`notification-button ${open ? "notification-button-open" : ""}`}
        icon={
          <Badge
            className="notification-badge"
            count={activities?.filter((act) => !act?.notification_read).length || 0}
          >
            <BellOutlined className="notification-icon" />
          </Badge>
        }
      />
      <div className={`notification-drawer ${open ? "notification-drawer-open" : ""}`}>
        <Tabs
          defaultActiveKey="1"
          tabBarStyle={{ backgroundColor: "#FFF" }}
          className="notification-tabs"
        >
          <Tabs.TabPane
            className="notification-tab-pane"
            tab={
              <Typography.Title level={5} className="notification-tab-header">
                Mine Activity
              </Typography.Title>
            }
            key="1"
          >
            <div className="notification-button-all-container">
              <Button
                className="notification-button-all"
                size="small"
                type="text"
                onClick={() => handleMarkAllAsRead()}
              >
                Mark all as read
              </Button>
            </div>
            {(activities || [])?.map((activity) => (
              <div className="notification-list-item" key={activity.notification_guid}>
                <div className={!activity.notification_read ? "notification-dot" : ""} />
                <div tabIndex={0} role="button" onClick={() => activityClickHandler(activity)}>
                  <Typography.Text>{activity.notification_document?.message}</Typography.Text>
                  <Row className="items-center margin-small" gutter={6}>
                    <Col>
                      <Typography.Text className="notification-info-text">
                        {activity.notification_document?.metadata?.mine?.mine_name}
                      </Typography.Text>
                    </Col>
                    {activity.notification_document?.metadata?.permit && (
                      <>
                        <Col>
                          <div className="notification-separator" />
                        </Col>
                        <Col>
                          <Typography.Text className="notification-info-text">
                            {activity.notification_document?.metadata?.permit?.permit_no}
                          </Typography.Text>
                        </Col>
                        <Col>
                          <div className="notification-separator" />
                        </Col>
                      </>
                    )}
                    <Col>
                      <Typography.Text className="notification-info-text">
                        {formatDateTime(activity.create_timestamp)}
                      </Typography.Text>
                    </Col>
                  </Row>
                </div>
              </div>
            ))}
          </Tabs.TabPane>
        </Tabs>
      </div>
    </div>
  );
};

export default NotificationDrawer;
