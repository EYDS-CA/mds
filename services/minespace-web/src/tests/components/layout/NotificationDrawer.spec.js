import React from "react";
import NotificationDrawer from "@/components/layout/NotificationDrawer";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import { AUTHENTICATION, ACTIVITIES } from "@mds/common/constants/reducerTypes";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "@/App";

const initialState = {
  [AUTHENTICATION]: {
    userInfo: { preferred_username: "USERNAME" },
  },
  [ACTIVITIES]: {
    activities: MOCK.ACTIVITIES.data.records,
  },
};

describe("Notification Drawer", () => {
  it("renders properly", () => {
    const { container } = render(
      <ReduxWrapper initialState={initialState}>
        <Provider store={store}>
          <NotificationDrawer />
        </Provider>
      </ReduxWrapper>
    );

    expect(container).toMatchSnapshot();
  });
});
