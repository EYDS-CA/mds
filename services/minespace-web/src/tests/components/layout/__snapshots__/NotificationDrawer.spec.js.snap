// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Drawer renders properly 1`] = `
<div>
  <div>
    <button
      class="ant-btn ant-btn-text ant-btn-icon-only notification-button "
      type="button"
    >
      <span
        class="ant-badge notification-badge"
      >
        <span
          aria-label="bell"
          class="anticon anticon-bell notification-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="bell"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
            />
          </svg>
        </span>
      </span>
    </button>
    <div
      class="notification-drawer "
    >
      <div
        class="ant-tabs ant-tabs-top notification-tabs"
      >
        <div
          class="ant-tabs-nav"
          role="tablist"
          style="background-color: rgb(255, 255, 255);"
        >
          <div
            class="ant-tabs-nav-wrap"
          >
            <div
              class="ant-tabs-nav-list"
              style="transform: translate(0px, 0px);"
            >
              <div
                class="ant-tabs-tab ant-tabs-tab-active"
                data-node-key="1"
              >
                <div
                  aria-controls="rc-tabs-test-panel-1"
                  aria-selected="true"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-1"
                  role="tab"
                  tabindex="0"
                >
                  <h5
                    class="ant-typography notification-tab-header"
                  >
                    Mine Activity
                  </h5>
                </div>
              </div>
              <div
                class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
          <div
            class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
            style="background-color: rgb(255, 255, 255);"
          >
            <button
              aria-controls="rc-tabs-test-more-popup"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-hidden="true"
              class="ant-tabs-nav-more"
              id="rc-tabs-test-more"
              style="visibility: hidden; order: 1;"
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="ant-tabs-content-holder"
        >
          <div
            class="ant-tabs-content ant-tabs-content-top"
          >
            <div
              aria-hidden="false"
              aria-labelledby="rc-tabs-test-tab-1"
              class="ant-tabs-tabpane ant-tabs-tabpane-active notification-tab-pane"
              id="rc-tabs-test-panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <div
                class="notification-button-all-container"
              >
                <button
                  class="ant-btn ant-btn-text ant-btn-sm notification-button-all"
                  type="button"
                >
                  <span>
                    Mark all as read
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
